From a78c6da0376bfb67560725aa5ea0423de1f6f7f9 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Mon, 11 May 2020 19:56:33 +0200
Subject: [PATCH] fsl-imx8mm-var-dart: Switch pinmux as requested

We do so according to the requirements doc at
revision from May 6.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 .../dts/freescale/fsl-imx8mm-var-dart.dts     | 22 +++++++++++++------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-var-dart.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-var-dart.dts
index e364863cd6b5..505d69c38ffd 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-var-dart.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-var-dart.dts
@@ -183,9 +183,23 @@
 
 &iomuxc {
 	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpios>;
 
 	imx8mm-var-dart {
 
+		pinctrl_gpios: gpiosgrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO00_GPIO1_IO0              0x140
+				MX8MM_IOMUXC_ENET_TX_CTL_GPIO1_IO22            0x140
+				MX8MM_IOMUXC_ENET_MDIO_GPIO1_IO17              0x140
+				MX8MM_IOMUXC_ENET_MDC_GPIO1_IO16               0x140
+				MX8MM_IOMUXC_GPIO1_IO11_GPIO1_IO11             0x140
+				MX8MM_IOMUXC_SAI3_TXD_GPIO5_IO1                0x140
+				MX8MM_IOMUXC_SAI3_TXC_GPIO5_IO0                0x140
+				MX8MM_IOMUXC_SPDIF_RX_GPIO5_IO4                0x140
+			>;
+		};
+
 		pinctrl_csi1: csi1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI1_RXD6_GPIO4_IO8		0x19
@@ -195,8 +209,6 @@
 
 		pinctrl_fec1: fec1grp {
 			fsl,pins = <
-				MX8MM_IOMUXC_ENET_MDC_ENET1_MDC			0x3
-				MX8MM_IOMUXC_ENET_MDIO_ENET1_MDIO		0x3
 				MX8MM_IOMUXC_ENET_TD3_ENET1_RGMII_TD3		0x1f
 				MX8MM_IOMUXC_ENET_TD2_ENET1_RGMII_TD2		0x1f
 				MX8MM_IOMUXC_ENET_TD1_ENET1_RGMII_TD1		0x1f
@@ -208,7 +220,6 @@
 				MX8MM_IOMUXC_ENET_TXC_ENET1_RGMII_TXC		0x1f
 				MX8MM_IOMUXC_ENET_RXC_ENET1_RGMII_RXC		0x91
 				MX8MM_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x91
-				MX8MM_IOMUXC_ENET_TX_CTL_ENET1_RGMII_TX_CTL	0x1f
 				MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x19
 				MX8MM_IOMUXC_GPIO1_IO07_GPIO1_IO7		0x41
 			>;
@@ -271,8 +282,6 @@
 				MX8MM_IOMUXC_SAI3_RXC_SAI3_RX_BCLK		0xd6
 				MX8MM_IOMUXC_SAI3_RXD_SAI3_RX_DATA0		0xd6
 				MX8MM_IOMUXC_SAI3_TXFS_SAI3_TX_SYNC		0xd6
-				MX8MM_IOMUXC_SAI3_TXC_SAI3_TX_BCLK		0xd6
-				MX8MM_IOMUXC_SAI3_TXD_SAI3_TX_DATA0		0xd6
 				MX8MM_IOMUXC_SAI3_MCLK_SAI3_MCLK		0xd6
 			>;
 		};
@@ -292,7 +301,6 @@
 		pinctrl_spdif1: spdif1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SPDIF_TX_SPDIF1_OUT		0xd6
-				MX8MM_IOMUXC_SPDIF_RX_SPDIF1_IN			0xd6
 			>;
 		};
 
@@ -893,7 +901,7 @@
 	phy-reset-gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
 	phy-reset-duration = <10>;
 	fsl,magic-packet;
-	status = "okay";
+	status = "disabled";
 
 	mdio {
 		#address-cells = <1>;
-- 
2.17.1

